ProcessLib/RichardsFlow/RichardsFlowFEM.h

```diff
@@ -177,13 +226,18 @@ class LocalAssemblerData : public RichardsFlowLocalAssemblerInterface
     {
         SpatialPosition pos;
         pos.setElementID(_element.getID());
-
-        auto const K = _process_data.intrinsic_permeability(t, pos)[0];
-        MathLib::PiecewiseLinearInterpolation const& interpolated_Pc =
-            _process_data.interpolated_Pc;
-        MathLib::PiecewiseLinearInterpolation const& interpolated_Kr =
-            _process_data.interpolated_Kr;
-        auto const mu = _process_data.viscosity(t, pos)[0];
+        const int material_id =
+            _process_data.material->getMaterialID(pos.getElementID().get());
```

Comment:
minor: boost::optional also has `operator*` for access. One also could use `_element.getID()` as it is written two lines above.